app/models/user.rb

```diff
@@ -0,0 +1,41 @@
+require 'bcrypt'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :username, String, unique: true, message: ""
+  property :first_name, String
+  property :last_name, String
+  property :email, String, unique: true, message: ""
+  property :password_digest, Text
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+
+  validates_confirmation_of :password, message:
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.